### PR TITLE
add(deepresearch)：add build interactive html function

### PR DESCRIPTION
**Pull request by Gfangxin**
### Describe what this PR does 
- Converts generated in-depth research reports into interactive HTML code using AI models.  

**Why we need it**  
- Enables dynamic visualization and user engagement with research findings (e.g., expandable sections, filters).  

### Describe how you did it
1. Built `InteractionAgent` with configured system prompts for HTML generation.  
2. Added `InteractiveHtmlController` endpoint:  
   - Fetches stored reports by `reportId`.  
   - Forwards content to AI for HTML conversion.  

### Describe how to verify it 
1. Generate a report via Deep Research.  
2. Call the API and check output:  
   ```bash  
   curl --location --request GET 'http://localhost:8080/api/interactive-html/buildHtml?reportId=xxxxx'  
   ```  
   - Validate HTML structure/interactivity.  
   - Test edge cases (empty reports, invalid IDs).  

### Special notes for reviews
🚧 **Future enhancements (non-blocking):**  
- Support multi-round iteration for HTML refinements based on user feedback.  
- Add persistent storage for generated HTML files.

### CODE REVIEW

File: spring-ai-alibaba-deepresearch/src/main/java/com/alibaba/cloud/ai/example/deepresearch/controller/InteractiveHtmlController.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+		log.info("InteractiveHtmlController initialized with ReportService");
+	}
+
+	/**
+	 * building an interactive html report(构建交互式HTML报告)
+	 * @param reportId 报告ID
+	 * @return Return a Flux stream containing events from the build
+	 * process(返回一个Flux流，包含构建过程中的事件)
+	 */
+	@RequestMapping(value = "/buildHtml", method = RequestMethod.GET, produces = MediaType.TEXT_EVENT_STREAM_VALUE)
```
Comment (GTyingzi) [inline spring-ai-alibaba-deepresearch/src/main/java/com/alibaba/cloud/ai/example/deepresearch/controller/InteractiveHtmlController.java]: 接口路径：/buildHtml -> /build吧

File: spring-ai-alibaba-deepresearch/src/main/java/com/alibaba/cloud/ai/example/deepresearch/controller/InteractiveHtmlController.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+ * @since 2025/6/20 交互式HTML报告控制器，提供生成和展示交互式HTML报告的API Interactive HTML Report Controller,
+ * providing APIs for generating and displaying interactive HTML reports
+ */
+@Controller
+@RequestMapping("/api/interactive-html")
+public class InteractiveHtmlController {
+
+	private static final Logger log = LoggerFactory.getLogger(InteractiveHtmlController.class);
+
+	private final ReportService reportService;
```
Comment (GTyingzi) [inline spring-ai-alibaba-deepresearch/src/main/java/com/alibaba/cloud/ai/example/deepresearch/controller/InteractiveHtmlController.java]: 一个自动注入，一个构造器注入不太规范。这里我们统一改外自动注入吧

File: spring-ai-alibaba-deepresearch/src/main/resources/prompts/buildInteractiveHtmlPrompt.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    </body>
+    </html>
+    ```
+4. Constraints
+   - All elements must be strictly derived from the original report
+   - Interaction design needs to directly address the issues raised in the report
+   - The code must be able to run directly in the browser
+
+5. Output Format:
+    only output the HTML code, do not output any other text.
```
Comment (GTyingzi) [inline spring-ai-alibaba-deepresearch/src/main/resources/prompts/buildInteractiveHtmlPrompt.md]: 这下面再留一个空白行，不然过不了check
